### PR TITLE
heatmap pipeline

### PR DESCRIPTION
**Pull request by nbokulich**
Added a new heatmap pipeline for creating abundance heatmaps of important features.

Eventually this will go into the main pipelines (e.g., `classify_samples`) but for now will leave it out for a release

### CODE REVIEW

File: q2_sample_classifier/tests/test_classifier.py [hunk 1]
```diff
@@ -870,6 +870,44 @@ def test_predict_feature_order_aint_no_thing(self):
...
+            self.table_ecam, self.imp, self.md_ecam, group_samples=True)
+        self.assertEqual(table.view(biom.Table).shape, (1056, 2))
+
+    def test_heatmap_importance_threshold(self):
+        heatmap, table, = sample_classifier.actions.heatmap(
+            self.table_ecam, self.imp, self.md_ecam,
+            importance_threshold=0.017)
+        self.assertEqual(table.view(biom.Table).shape, (10, 121))
+
+    # def test_heatmap_feature_count(self):
```
Comment (thermokarst) [inline q2_sample_classifier/tests/test_classifier.py]: Is this supposed to be commented out? Same with L905.

Comment (nbokulich) [reply q2_sample_classifier/tests/test_classifier.py]: crap... thanks for catching (those were commented out while I was building tests). Have fixed in latest commit.

File: q2_sample_classifier/plugin_setup.py [hunk 2]
```diff
@@ -529,6 +530,55 @@
...
+        'method': 'Clustering methods exposed by seaborn (see http://seaborn.'
+                  'pydata.org/generated/seaborn.clustermap.html#seaborn.clust'
+                  'ermap for more detail).',
+        'cluster': 'Specify which axes to cluster.',
+        'color_scheme': 'Color scheme for heatmap.',
+    },
+    output_descriptions={
+        'heatmap': 'Heatmap of important features.',
+        'filtered_table': 'Filtered feature table containing data displayed '
+                          'in your lovely new heatmap.'},
```
Comment (thermokarst) [inline q2_sample_classifier/plugin_setup.py]: There is no "love" in bioinformatics.

File: q2_sample_classifier/classify.py [hunk 3]
```diff
@@ -350,6 +350,52 @@ def summarize(output_dir: str, sample_estimator: Pipeline):
     _summarize_estimator(output_dir, sample_estimator)
 
 
+def heatmap(ctx, table, importance, metadata=None, feature_count=50,
+            importance_threshold=0, group_samples=True, normalize=True,
```
Comment (thermokarst) [inline q2_sample_classifier/classify.py]:
```suggestion
            importance_threshold=0, group_samples=False, normalize=True,
```

Thoughts?

File: q2_sample_classifier/classify.py [hunk 4]
```diff
@@ -350,6 +350,52 @@ def summarize(output_dir: str, sample_estimator: Pipeline):
...
+            importance_threshold=0, group_samples=False, normalize=True,
+            metric='braycurtis', method='average', cluster='features',
+            color_scheme='rocket'):
+    filter_features = ctx.get_action('feature_table', 'filter_features')
+    group = ctx.get_action('feature_table', 'group')
+    make_heatmap = ctx.get_action('feature_table', 'heatmap')
+
+    if group_samples and metadata is None:
+        raise ValueError(
+            'If group_samples is True, you must provide metadata.')
```
[the remark this answers is not in the document]
Comment (nbokulich) [reply q2_sample_classifier/classify.py]: 👍

File: q2_sample_classifier/plugin_setup.py [hunk 5]
```diff
@@ -529,6 +530,55 @@
...
+    name='Generate heatmap of important features.',
+    description='Generate a heatmap of important features. Features are '
+                'filtered based on importance scores; samples are optionally '
+                'grouped by metadata; and a lovely new heatmap is born.'
```
[the remark this answers is not in the document]
Comment (nbokulich) [reply q2_sample_classifier/plugin_setup.py]: 👍

File: q2_sample_classifier/plugin_setup.py (hunk 2 above)
Comment (nbokulich) [reply q2_sample_classifier/plugin_setup.py]: 😦

File: q2_sample_classifier/classify.py (hunk 3 above)
Comment (nbokulich) [reply q2_sample_classifier/classify.py]: lgtm, already made change in my local branch.

**Comment by nbokulich** [comment]
thanks @thermokarst ! Fixed all requests, tests pass locally, back to you.

**Pull request merged**